cf/samples/javascript-samples/sap-iots-device-rest/index.js

```diff
@@ -0,0 +1,163 @@
+     *     the properties defined by the Capability. If they are objects, the keys must be strings that match
+     *     the names of the properties, and the values must have the types specified for the properies
+     *     defined by the Capability.
+     */
+    async function postMeasurementData(deviceAlternateId, sensorAlternateId, capabilityAlternateId, measurements) {
+        if (!Array.isArray(measurements)) {
+            measurements = [measurements];
+        }
+        for (const measurement of measurements) {
+            if (typeof measurement !== 'object') {
+                throw new TypeError('measurements must be an object or an array of objects');
+            }
+        }
+        if (typeof deviceAlternateId !== 'string') {
+            throw new TypeError('deviceAlternateId must be a string');
+        }
+        if (typeof sensorAlternateId !== 'string') {
+            throw new TypeError('sensorAlternateId must be a string');
+        }
+        if (typeof capabilityAlternateId !== 'string') {
+            throw new TypeError('capabilityAlternateId must be a string');
+        }
+        
+        const payload = {
+            'capabilityAlternateId': capabilityAlternateId,
+            'sensorAlternateId': sensorAlternateId,
+            'measures': measurements
+        };
+
+        const body = JSON.stringify(payload);
+
+        let promise = new Promise(function(resolve, reject) {
+            const httpsOptions = {
+                hostname: params.hostname,
+                port: 443,
+                path: '/iot/gateway/rest/measures/' + deviceAlternateId,
+                method: 'POST',
+                headers: {'Content-Type': 'application/json'},
+                key: params.clientPrivateKeyPEM,
+                passphrase: params.passphrase,
+                cert: params.clientCertPEM,
+                agent: httpsAgent
+            };
+
+            const request = https.request(httpsOptions, (response) => {
+                console.debug('REST gateway response code: %d', response.statusCode);
+                response.setEncoding('utf8');
+                let responseBody = '';
+                response.on('data', (chunk) => {
+                    responseBody += chunk;
+                    console.debug('Got chunk of HTTP response:', chunk);
+                });
+                response.on('end', () => {
+                    console.debug('End of HTTP response.');
+                    if (response.statusCode === 200 || response.statusCode === 202) {
+                        resolve(response.statusCode);
+                    } else {
+                        let errorMessages = '';
+                        try {
+                            const resp = JSON.parse(responseBody);
+                            console.debug(resp);
+                            errorMessages = resp[0].messages.join('\n');
```

Comment:
why don't we check it exists?

Reply:
Added comments: this is error handling already, so we ignore all the secondary errors that may happen when we extracting the error messages.